content/docs/connect/choose-connection.md

```diff
@@ -0,0 +1,163 @@
+---
+title: Choosing your connection type
+subtitle: Rules of thumb for choosing one connection type over another
+enableTableOfContents: true
+---
+
+When setting up your application’s connection to your Neon Postgres database, you need to make two main choices: the type of driver and the type of connection. This flowchart will guide you through these options.
+
+## Choosing your connection type: flowchart
+
+![choose your connection type](/docs/connect/choose_connection.png)
+
+## Choosing your connection type: drivers and pooling
+
+### Step 1: Choose your Postgres driver
+
+Your first choice is which driver to use. If working in a serverless environment, use the [Neon Serverless Driver](/docs/serverless/serverless-driver). It's built to handle dynamic workloads with high variability in traffic &#8212; for example, Vercel Edge Functions or Cloudflare Workers.
+
+If you don't need serverless, use a traditional TCP-based Postgres driver. For example, if you’re using Node.js with a framework like Next.js, you can add the `pg` client to your dependencies, which acts as the TCP-based Postgres driver.
+
+### Step 2: If serverless, choose HTTP or WebSocket connections
+
+Neon offers two types of serverless connections: HTTP and WebSocket.
+
+- **Use HTTP for**: One-shot queries and short-lived operations where connection setup and teardown costs are minimal.
+- **Use WebSocket for**: [Transactions](/docs/postgresql/query-reference#transactions) and operations requiring persistent connections over a session. WebSocket connections provide lower latency for multiple queries in the same session.
```

Comment:
```suggestion
- **Use WebSocket for**: [Transactions](/docs/postgresql/query-reference#transactions) and operations requiring persistent connections over a session. WebSocket connections provide lower latency for multiple queries in the same session and are fully compatible with [node-postgres](https://node-postgres.com/).
```

Reply:
Not sure about the "provide lower latency for multiple queries in the same session". Lower than HTTP or lower than TCP connections? HTTP connections support batched queries (a batch of one-shot queries), but not queries within a transaction. so this statement might be confusing.

Reply:
double-check "lower latency for multiple queries"

Reply:
I modified to just use the same text as the serverless page: unless it can be simplified, why re-invent the wheel